Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -66,6 +66,40 @@
     _gTracersToProfilers[tracerKey] = profiler;
 }
 
+void
+discardProfilerForTracer(SentryTracer *tracer)
+{
+    std::lock_guard<std::mutex> l(_gStateLock);
+
+    SENTRY_CASSERT(_gTracersToProfilers != nil && _gProfilersToTracers != nil,
+        @"Structures should have already been initialized by the time they are being queried");
+
+    const auto tracerKey = tracer.traceId.sentryIdString;
+    const auto profiler = _gTracersToProfilers[tracerKey];
+
+    if (!SENTRY_CASSERT_RETURN(profiler != nil,
+            @"Expected a profiler to be associated with tracer id %@.", tracerKey)) {
```

Comment:
`m`: I'm unsure if we need to abort in debug here. We could exit if there is no profiler for the tracer.

Reply:
`profilerForFinishedTracer` is called from `-[SentryProfiler createProfilingEnvelopeItemForTransaction:]` which should only be called if the tracer is being profiled, so otherwise there would be a logic error somewhere, so I added this assert to help catch it if it ever happens.